x-pack/plugins/security_solution/public/common/hooks/flyout/use_init_flyout_url_param.ts

```diff
@@ -39,16 +39,28 @@ export const useInitFlyoutFromUrlParam = () => {
   }, []);
 
   const loadExpandedDetailFromUrl = useCallback(() => {
-    const { initialized, isLoading, totalCount } = dataTableCurrent;
+    const { initialized, isLoading, totalCount, additionalFilters } = dataTableCurrent;
     const isTableLoaded = initialized && !isLoading && totalCount > 0;
-    if (urlDetails && isTableLoaded) {
-      updateHasLoadedUrlDetails(true);
-      dispatch(
-        dataTableActions.toggleDetailPanel({
-          id: TableId.alertsOnAlertsPage,
-          ...urlDetails,
-        })
-      );
+    if (urlDetails) {
+      if (!additionalFilters.showBuildingBlockAlerts) {
```

Comment:
This is a great solution. Do you also think, we can enable `buildingblockAlerts` generally because then it will resolve the issues with redirection from `dashboard` -> `Alerts Page`.

Paul seemed okay with it, right?

Reply:
Yea. This works for redirects when the flyout is opened, but best way to do it outside of this scenario would be to initialize it in the redux store to true when the table is loaded.

Reply:
Cool. will make that change.